### ISSUE TITLE
mmx silently ignored in function argument

### ISSUE THREAD

**Issue by eponier** [jasmin-lang/jasmin#652]
This program exhibits the problem.
```
fn f (#mmx reg u64 y) -> reg u64 {
  reg u64 x;
  x = 0;
  return x;
}

export fn main () -> reg u64 {
  reg u64 x;
  x=0;
  x = f(x);
  return x;
}
```
`y` is compiled as `RAX`. This works because `y` is not used.

**Comment by MrDaiki** [comment]
A little question here : what is the compilation pass where register used for variables are computed (I think it is reg-alloc but I would prefer to be sure)

**Comment by vbgl** [comment]
Yes: the compiler step mapping variables to registers is called register allocation. However, this is unrelated to this issue: here there is a “kind” mismatch between the parameter and the argument that is silently ignored: this could be detected early by a simple check.
